addons/point_of_sale/static/src/js/Screens/TicketScreen/TicketScreen.js

```diff
@@ -305,11 +305,20 @@ odoo.define('point_of_sale.TicketScreen', function (require) {
                 return this._getOrderStates().get(this._getScreenToStatusMap()[screen.name]).text;
             }
         }
+        /**
+         * If the order is the only order and is empty
+         */
+        isDefaultOrderEmpty(order) {
+            let status = this._getScreenToStatusMap()[order.get_screen_data().name];
+            let productScreenStatus = this._getScreenToStatusMap().ProductScreen;
+            return order.get_orderlines().length === 0 && this.env.pos.get_order_list().length === 1 && status === productScreenStatus;
```

Comment:
you should also check order.get_payment_lines

Reply:
Yeah indeed, if the order is empty but has some payments that are not electronic and validated, it can be deleted